### PR TITLE
Allow an empty PUT and POST to keep compatibility

### PR DESCRIPTION
**Pull request by nfukuzaki**
This PR is related to https://github.com/square/okhttp/pull/1559

### CODE REVIEW

File: okhttp-apache/src/test/java/com/squareup/okhttp/apache/OkApacheClientTest.java [hunk 1]
```diff
@@ -113,6 +117,16 @@
     assertEquals("Hello, world!", request.getBody().readUtf8());
     assertEquals(request.getHeader("Content-Length"), "13");
   }
+  @Test public void postEmptyEntity() throws Exception {
+    server.enqueue(new MockResponse());
+    final HttpPost post = new HttpPost(server.getUrl("/").toURI());
+    client.execute(post);
```
Comment (JakeWharton) [inline okhttp-apache/src/test/java/com/squareup/okhttp/apache/OkApacheClientTest.java]: These tests don't assert anything. Copy the assertions from another test and ensure the body was indeed missing.

File: okhttp-apache/src/test/java/com/squareup/okhttp/apache/OkApacheClientTest.java [hunk 2]
```diff
@@ -3,19 +3,23 @@
 import com.squareup.okhttp.mockwebserver.MockResponse;
 import com.squareup.okhttp.mockwebserver.MockWebServer;
 import com.squareup.okhttp.mockwebserver.RecordedRequest;
+
```
Comment (swankjesse) [inline okhttp-apache/src/test/java/com/squareup/okhttp/apache/OkApacheClientTest.java]: PROTIP: if you want to win the hearts & minds of open source developers, don't make unrelated formatting changes when you submit a pull request!

File: okhttp-apache/src/main/java/com/squareup/okhttp/apache/OkApacheClient.java [hunk 3]
```diff
@@ -67,6 +69,8 @@ private static Request transformRequest(HttpRequest request) {
         if (encoding != null) {
           builder.header(encoding.getName(), encoding.getValue());
         }
+      } else {
```
Comment (swankjesse) [inline okhttp-apache/src/main/java/com/squareup/okhttp/apache/OkApacheClient.java]: When would a user of Apache HTTP client return a null entity? Isn't that a bug?

Comment (nfukuzaki) [inline okhttp-apache/src/main/java/com/squareup/okhttp/apache/OkApacheClient.java]: Apache HttpClient is able to send POST/PUT request without entity body.
For example following code works properly.

```java
HttpPut put = new HttpPut("URL");
HttpClient client = new DefaultHttpClient();
client.execute(put);
```

But OkApacheClient(master blanch) will throw IllegalArgumentException.
java.lang.IllegalArgumentException: method PUT must have a request body.

```java
HttpPut put = new HttpPut("URL");
OkApacheClient client = new OkApacheClient();
client.execute(put);
```

File: okhttp-apache/src/test/java/com/squareup/okhttp/apache/OkApacheClientTest.java (hunk 2 above)
Comment (nfukuzaki) [inline okhttp-apache/src/test/java/com/squareup/okhttp/apache/OkApacheClientTest.java]: Thank you for your advice.
I will fix it.

**Comment by swankjesse** [comment]
Thank you! I rebased & squashed & merged this.
